connection/mysql_utils.go

```diff
@@ -432,3 +432,24 @@ type Row interface {
 }
 
 type RowFactory func() Row
+
+func ChunkRows(done <-chan struct{}, rows []Row, size int) <-chan []Row {
+	ch := make(chan []Row)
+
+	go func() {
+		defer close(ch)
+		for i := 0; i < len(rows); i += size {
+			end := i + size
+			if end > len(rows) {
+				end = len(rows)
+			}
+			select {
+			case ch <- rows[i:end]:
+			case <-done:
+				return
+			}
+		}
+	}()
+
+	return ch
+}
```

Comment:
How many rows you expect so you don't compute and return an `[][]Row` sync-ly?

Reply:
This can get really large, depending on the size of your custom vars. 
I'm not sure, if this helps a lot, but I also don't think it'll hurt us. What do you think?

Reply:
**How** large in the worst case?

Reply:
Basically infinite, because we don't have limits on how large a custom var can be, right?

Reply:
The point is that you already have a slice `rows[]Row`. The `[][]Row` I suggest would be about `len(rows)/500` long and its `[]Row` items would re-use `rows[i][j]`'s memory.